src/xdocs/config_coding.xml

```diff
@@ -3130,6 +3130,38 @@ public class Test {
     return new TreeSet&lt;Foo&lt;Boolean&gt;&gt;();
   }
 
+}
+        </source>
+        <p>
+            To configure the check so that it limit IllegalType to validate field types only:
+        </p>
+        <source>
+&lt;module name="IllegalType"&gt;
+  &lt;property name=&quot;illegalClassNames&quot; value=&quot;java.util.Optional&quot;/&gt;
+  &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
+&lt;/module&gt;
+&lt;module name="SuppressionXpathSingleFilter"&gt;
+  &lt;property name=&quot;query&quot; value=&quot;//METHOD_DEF//*&quot;/&gt;
+&lt;/module&gt;
+        </source>
+        <source>
+import java.util.Optional;
+
+class Optional {
+
+  private Optional&lt;String&gt; field1; // violation, usage of type 'Optional' is not allowed
+  static int var = 4; // OK
+  protected notOptional&lt;String&gt; field2; // OK
+  public Optional&lt;String&gt; field3; // violation, usage of type 'Optional' is not allowed
+  Optional&lt;String&gt; field4; // violation, usage of type 'Optional' is not allowed
+
+  void foo() {
+    Optional&lt;String&gt; i; // OK
+  }
+
+  public &lt;T extends java.util.Optional&gt; void Optionalmethod(T t){
```

Comment:
Same

Reply:
changes  `Optionalmethod` to `Method`.

Reply:
how did you know about this?

Reply:
Go to Details of that failed check, and click view more details in azure pipelines. You would find the details there.

Reply:
i try that but didn't found it. but it should pass all checks this time. right?

Reply:
i try that and found the spell error but didn't specifically found it on Optionalmethod may i missed it. but it should pass all checks this time.

Reply:
i try that and found the spell error but didn't specifically found it on Optionalmethod maybe i missed it. but it should pass all checks this time.

Reply:
i tried that and found the spell error but didn't specifically found it on Optionalmethod maybe i missed it. but it should pass all checks this time.
was the error because of `Camel Casing in java`?

Reply:
If word is not an english word, then it must be present in [jsoref-spellchecker/whitelist.words](https://github.com/checkstyle/checkstyle/blob/master/.ci/jsoref-spellchecker/whitelist.words). Camel Case would not give this error.